modules/core/arrow-instances-core/src/main/kotlin/arrow/instances/number.kt

```diff
@@ -2,150 +2,234 @@ package arrow.instances
 
 import arrow.typeclasses.*
 
-class NumberSemigroup<A : Number>(val f: (A, A) -> A) : Semigroup<A> {
-  override fun A.combine(b: A): A = f(this, b)
-}
-
 //////////
 // Byte
 //////////
-
-private val SGByte: Semigroup<Byte> = NumberSemigroup({ one, two -> (one + two).toByte() })
-
-object ByteMonoidInstance : Monoid<Byte>, Semigroup<Byte> by SGByte {
-  override fun empty(): Byte = 0
+interface ByteSemigroupInstance : Semigroup<Byte> {
```

Comment:
does it make sense to call instances to interfaces that you still need to instantiate ? why not returning the objects already implementing those anymore?

Reply:
This is the same convention that is used across `Arrow`. Originally the sufix `Instance` was added to have a naming convention by which to look up instances at runtime. Since 0.7.0 eliminates the runtime system I propose that we get rid of `Instances` sufixing names across the board. the `OptionFunctorInstance` becomes `OptionFunctor` and it's real instantiated instance is:
```
interface OptionFunctor {
  ...
  companion object {
    val defaultInstace: Functor<ForOption> = object : OptionFunctor {}  
  }
}

fun Option.Companion.functor(): Functor<ForOption> = 
   OptionFunctor.defaultInstance
```
This brings us simpler names and also proper singleton values for instances that require no further arguments for instantiation. If this looks good we should tackle this refactoring in a separate PR.

Reply:
Sounds good to me, but we would probably want to discuss that on an issue apart to avoid blocking here and move towards 0.7.0 faster for monday. (since that doesn't sound blocker at all anyways) There you go #767